### DISCUSSION TITLE
How do I make a folder?

### DISCUSSION THREAD

**Discussion by TheJasonPlayz** [community/community#21705] [New to GitHub]
<p>So I would like to use Github as a place to store my python programs, and I want them organized while they are here. But I cannot find how to make a folder to keep them sorted, can I have some help?</p>
<p>Jason</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>git doesn’t have a concept of “folders” by itself. Git tracks files and if a file happens to have a name like <code>foo/bar/baz.txt</code>, then when git recreates the repository locally it creates the <code>foo</code> folder, the <code>foo/bar</code> folder, and then the <code>baz.txt</code> file within the <code>foo/bar</code> folder.</p>
<p>So if you want to create a folder within a repository, you have to create a file within it. One common way people do this is to create an empty file named <code>.gitkeep</code> in the directory that they want to exist.</p>
<p>To create a folder this way through the GitHub web interface, you would follow <a href="https://help.github.com/en/articles/creating-new-files" rel="nofollow noopener">the instructions for creating files</a>. When it comes time to name the file, you would enter <code>foo/bar/.gitkeep</code>. You’ll notice that each time you press the <code>/</code> key, the GitHub UI recognizes that you mean to create a folder. Once you finish following the instructions, you’ll see that the folders and file exist, like I did <a href="https://github.com/lee-dohm/test-repo/tree/master/foo/bar" rel="nofollow noopener">on my test repo</a>.</p>
<p>I hope that helps!</p>
